src/xtgeo/grid3d/_gridprop_import_eclrun.py

```diff
@@ -536,50 +534,172 @@ def import_gridprops_from_restart_file_sections(
+
+    Note: Does not check that all names are found.
 
     Args:
         restart_filelike: The restart file.
         names: List of property names to be imported. Can also,
             be set to "all" to import all parameters.
         dates: List of xtgeo style dates (e.g. int(19990101)), can also
             be "all", "last" and "first".
-        grid: The grid used by the simulator to produce the init file.
+        grid: The grid used by the simulator to produce the restart file.
         fracture: If a dual porosity module, indicates that the fracture
             (as apposed to the matrix) grid property should be imported.
     Returns:
         List of GridProperty parameters matching the names and dates.
     """
     close = False
-    try:
-        if fformat == "unrst":
+    if isinstance(restart_filelike, (pathlib.Path, str)):
+        if fformat == eclio.Format.UNFORMATTED:
             filehandle = open(restart_filelike, "rb")
             close = True
-        elif fformat == "funrst":
+        elif fformat == eclio.Format.FORMATTED:
             filehandle = open(restart_filelike, "rt")
             close = True
         else:
             raise ValueError(f"Unsupported restart file format {fformat}")
-    except TypeError:
+    else:
+        # If restart_filelike is not a filename/path we assume
+        # its a stream
         filehandle = restart_filelike
         close = False
-    generator = section_generator(filter_lgr(eclio.lazy_read(filehandle)))
-    read_properties = import_gridprops_from_restart_file_sections(
-        generator,
-        names,
-        dates,
+    try:
+        generator = section_generator(filter_lgr(eclio.lazy_read(filehandle)))
+        read_properties = find_gridprops_from_restart_file_sections(
+            generator,
+            names,
+            dates,
+            grid,
+            fracture,
+        )
+    finally:
+        if close:
+            filehandle.close()
+    return read_properties
+
+
+def import_gridprop_from_init(pfile, name, grid, fracture=False):
+    """Import one parameter with the given name from an init file.
+
+    Args:
+        pfile: The init file.
+        name: The name of the parmaeter
+        grid: The grid used by the simulator to produce the init file.
+        fracture: If a dual porosity module, indicates that the fracture
+            (as apposed to the matrix) grid property should be imported.
+    Raises:
+        ValueError: If the parameter does not exist in the file.
+    Returns:
+        GridProperty parameter dictionary.
+    """
+    init_props = find_gridprop_from_init_file(pfile.file, [name], grid, fracture)
+    if len(init_props) != 1:
+        raise ValueError(f"Could not find property {name} in {pfile}")
+    init_props[0]["name"] = decorate_name(
+        init_props[0]["name"], grid.dualporo, fracture
+    )
+    return init_props[0]
+
+
+def sanitize_date(
+    date: Union[int, str, Literal["first", "last"]]
```

Comment:
The way I read this type is that we allow any int and any string and that mypy would not complain if we passed say "norway" as the date parameter.

What if we specified `Union[int, Literal["first", "last"]]` and have the caller be responsible for converting a string representation of a date to int?

Reply:
That is unfortunately possible now as what we are doing here is sanitizing input from the public API where str has been allowed.